### PR TITLE
RPM local repo support

### PR DESCRIPTION
**Pull request by alexhung**
Also add setting Artifactory custom base URL at `testAccPreCheck` so `TestAccRemoteCargoRepository` will pass.

### CODE REVIEW

File: pkg/artifactory/provider_test.go [hunk 1]
```diff
@@ -58,10 +60,17 @@ func getTestResty(t *testing.T) *resty.Client {
 }
 
 func testAccPreCheck(t *testing.T) {
-	// https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-UpdateCustomURLBase
 	restyClient := getTestResty(t)
+
+	// Set customer base URL so repos that relies on it will work
+	// https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-UpdateCustomURLBase
+	_, err := restyClient.R().SetBody(os.Getenv("ARTIFACTORY_URL")).SetHeader("Content-Type", "text/plain").Put("/artifactory/api/system/configuration/baseUrl")
```
Comment (chb0github) [inline pkg/artifactory/provider_test.go:67]: ?? What happened here?

Comment (chb0github) [reply pkg/artifactory/provider_test.go:67]: OOH!!! I know - yeah, I talked about this with Daniel.

File: pkg/artifactory/resource_artifactory_local_repository_test.go [hunk 2]
```diff
@@ -98,7 +98,7 @@ func TestAccLocalDebianRepository(t *testing.T) {
 			alias = "foo-alias{{ .kp_id }}"
 			private_key = <<EOF
 		-----BEGIN PGP PRIVATE KEY BLOCK-----
-		
+
```
Comment (chb0github) [inline pkg/artifactory/resource_artifactory_local_repository_test.go:101]: Did these tests run after the change? These keys are super sensitive to spaces and new lines

File: pkg/artifactory/resource_artifactory_local_rpm_repository.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+			Description: "A list of XML file names containing RPM group component definitions. Artifactory includes " +
+				"the group definitions as part of the calculated RPM metadata, as well as automatically generating a " +
+				"gzipped version of the group files, if required.",
+		},
+	})
+
+	type RpmLocalRepositoryParams struct {
+		LocalRepositoryBaseParams
+		RootDepth               int    `hcl:"yum_root_depth" json:"yumRootDepth"`
+		CalculateYumMetadata    bool   `hcl:"calculate_yum_metadata" json:"calculateYumMetadata"`
```
Comment (chb0github) [inline pkg/artifactory/resource_artifactory_local_rpm_repository.go:48]: I got that code done. Maybe you can give it a try in a few?

**Review by chb0github — COMMENTED** [review]
Just a few questions

File: pkg/artifactory/resource_artifactory_local_repository_test.go (hunk 2 above)
Comment (alexhung) [reply pkg/artifactory/resource_artifactory_local_repository_test.go:101]: It does IIRC. I will double check.

**Review by chb0github — APPROVED** [review]

File: pkg/artifactory/resource_artifactory_local_repository_test.go (hunk 2 above)
Comment (alexhung) [reply pkg/artifactory/resource_artifactory_local_repository_test.go:101]: I'm gonna revert the extra newlines. Look wrong even if they work.

**Pull request merged**